plugin/keyring_vault/vault_curl.h

```diff
@@ -0,0 +1,52 @@
+#ifndef MYSQL_VAULT_CURL_H
+#define MYSQL_VAULT_CURL_H
+
+#include <my_global.h>
+#include <curl/curl.h>
```

Comment:
My point is that  currently, any file which includes `vault_curl.h` will implicitly include `curl/curl.h` which is not a trivial one and in turn will include a bunch of unnecessary system headers like network stack. This can be avoided with minimal efforts.